src/components/PullRequestList/index.tsx

```diff
@@ -20,88 +20,83 @@ type PullRequestListProps = {
 
 const PullRequestList: FC<PullRequestListProps> = ({ prType }) => {
   const [pullRequests, setPullRequests] = useState<pullRequestType[]>([]);
+  const [noData, setNoData] = useState(false);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState('');
   const [page, setPage] = useState(1);
+  const [isBottom, setIsBottom] = useState(false);
 
   const prUrl = `${process.env.NEXT_PUBLIC_BASE_URL}/pullrequests/${prType}?page=${page}`;
 
   const fetchPRs = async () => {
     try {
       setLoading(true);
       const response = await axios.get(prUrl);
-      setPullRequests(response.data.pullRequests);
-
       if (!response.data.pullRequests.length) {
-        setError(`No more ${prType} PRs...`);
+        setNoData(true);
+      }
+      if (page > 1) {
+        setPullRequests((pullRequest) => [...pullRequest, ...response.data.pullRequests]);
       } else {
-        setError('');
+        setPullRequests(response.data.pullRequests);
       }
     } catch (err) {
       setPullRequests([]);
       setError('Error fetching pull requests!');
     } finally {
       setLoading(false);
+      setIsBottom(false);
     }
   };
-
-  const handlePrev = () => {
-    setPage(page > 1 ? page - 1 : 1);
-  };
-
-  const handleNext = () => {
-    setPage(pullRequests.length ? page + 1 : page);
-  };
-
   useEffect(() => {
     fetchPRs();
   }, [page]);
 
-  const getPRs = () => pullRequests.map((pullRequest: pullRequestType) => {
-    const {
-      title, username, createdAt, updatedAt, url: link,
-    } = pullRequest;
-    return (
-      <PullRequest
-        key={link}
-        title={title}
-        username={username}
-        createdAt={createdAt}
-        updatedAt={updatedAt}
-        url={link}
-      />
-    );
-  });
-
+  useEffect(() => {
+    if (isBottom && !noData) {
+      setPage(page + 1);
+    }
+  }, [isBottom]);
+  useEffect(() => {
+    const onScroll = () => {
+      const scrollTop = (document.documentElement
+        && document.documentElement.scrollTop)
+        || document.body.scrollTop;
+      const scrollHeight = (document.documentElement
+          && document.documentElement.scrollHeight)
+          || document.body.scrollHeight;
+      if (scrollTop + window.innerHeight + 50 >= scrollHeight) {
+        setIsBottom(true);
+      }
+    };
+    window.addEventListener('scroll', onScroll);
+    return () => window.removeEventListener('scroll', onScroll);
```

Comment:
Why two?

Reply:
Removing the event handler to prevent any memory leak. If somebody using older browser.

Reply:
Oh my bad, I thought it was 2 `addEventListeners` 😶